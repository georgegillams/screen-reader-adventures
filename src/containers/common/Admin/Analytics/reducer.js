import produce from 'immer';

import { load } from './actions';

export const initialState = {
  analytics: null,
  loading: false,
  loadError: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case load.REQUEST:
        draft.loading = true;
        draft.loadError = null;
        break;

      case load.SUCCESS:
        draft.loading = false;
        draft.analytics = payload;
        break;

      case load.FAILURE:
        draft.loading = false;
        draft.loadError = payload;
        break;
    }
  });

export default reducer;
