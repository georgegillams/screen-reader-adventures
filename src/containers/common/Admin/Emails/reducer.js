import produce from 'immer';

import { load, resend } from './actions';

export const initialState = {
  emails: null,
  loading: false,
  loadError: null,

  emailToResend: null,
  resending: false,
  resendError: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case load.REQUEST:
        draft.loading = true;
        draft.loadError = null;
        break;

      case load.SUCCESS:
        draft.loading = false;
        draft.emails = payload;
        break;

      case load.FAILURE:
        draft.loading = false;
        draft.loadError = payload;
        break;

      case resend.TRIGGER:
        draft.emailToResend = payload.emailToResend;
        break;

      case resend.REQUEST:
        draft.resending = true;
        draft.resendError = null;
        break;

      case resend.SUCCESS:
        draft.resending = false;
        break;

      case resend.FAILURE:
        draft.resending = false;
        draft.resendError = payload;
        break;
    }
  });

export default reducer;
