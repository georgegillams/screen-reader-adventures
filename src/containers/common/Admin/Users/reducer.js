import produce from 'immer';

import { load, remove, requestMagicLink, create, update } from './actions';

export const initialState = {
  users: null,
  loading: false,
  loadError: null,

  userToRemove: null,
  removing: false,
  removeError: null,

  userToLogIn: null,
  requesting: false,
  requestError: null,

  userToCreate: null,
  creating: false,
  createError: null,

  userToUpdate: null,
  updating: false,
  updateError: null,
  onUpdateSuccessCb: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case load.REQUEST:
        draft.loading = true;
        draft.loadError = null;
        break;

      case load.SUCCESS:
        draft.loading = false;
        draft.users = payload;
        break;

      case load.FAILURE:
        draft.loading = false;
        draft.loadError = payload;
        break;

      case remove.TRIGGER:
        draft.userToRemove = payload;
        break;

      case remove.REQUEST:
        draft.removing = true;
        draft.removeError = null;
        break;

      case remove.SUCCESS:
        draft.removing = false;
        break;

      case remove.FAILURE:
        draft.removing = false;
        draft.removeError = payload;
        break;

      case requestMagicLink.TRIGGER:
        draft.userToLogIn = payload;
        break;

      case requestMagicLink.REQUEST:
        draft.requesting = true;
        draft.requestError = null;
        break;

      case requestMagicLink.SUCCESS:
        draft.requesting = false;
        break;

      case requestMagicLink.FAILURE:
        draft.requesting = false;
        draft.requestError = payload;
        break;

      case create.TRIGGER:
        draft.userToCreate = payload;
        break;

      case create.REQUEST:
        draft.creating = true;
        draft.createError = null;
        break;

      case create.SUCCESS:
        draft.creating = false;
        break;

      case create.FAILURE:
        draft.creating = false;
        draft.createError = payload;
        break;

      case update.TRIGGER:
        draft.userToUpdate = payload.userToUpdate;
        draft.onUpdateSuccessCb = payload.onUpdateSuccessCb;
        break;

      case update.REQUEST:
        draft.updating = true;
        draft.updateError = null;
        break;

      case update.SUCCESS:
        draft.updating = false;
        break;

      case update.FAILURE:
        draft.updating = false;
        draft.updateError = payload;
        break;
    }
  });

export default reducer;
