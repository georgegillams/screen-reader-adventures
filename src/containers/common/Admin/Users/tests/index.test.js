import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { initialState } from '../reducer';
import { initialState as initialAuthenticatorState } from 'containers/common/Authenticator/reducer';

import configureStore from 'client-utils/common/redux/configure-store';

import AdminUsersIndex from '../index';
import AdminUsers from '../Container';

describe('<AdminUsers />', () => {
  let store;
  const spy = jest.fn();

  beforeAll(() => {
    store = configureStore({});
  });

  it('should render correctly - index', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminUsersIndex />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with initial state', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminUsers
          load={spy}
          remove={spy}
          requestMagicLink={spy}
          create={spy}
          update={spy}
          adminUsersState={{
            ...initialState,
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with user', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminUsers
          load={spy}
          remove={spy}
          requestMagicLink={spy}
          create={spy}
          update={spy}
          adminUsersState={{
            ...initialState,
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'testUser' },
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with admin user', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminUsers
          load={spy}
          remove={spy}
          requestMagicLink={spy}
          create={spy}
          update={spy}
          adminUsersState={{
            ...initialState,
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'testUser', admin: true },
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with loading=true', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminUsers
          load={spy}
          remove={spy}
          requestMagicLink={spy}
          create={spy}
          update={spy}
          adminUsersState={{
            ...initialState,
            loading: true,
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'testUser', admin: true },
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with loadError', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminUsers
          load={spy}
          remove={spy}
          requestMagicLink={spy}
          create={spy}
          update={spy}
          adminUsersState={{
            ...initialState,
            loadError: { error: 'some_error', errorMessage: 'Something went wrong' },
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'testUser', admin: true },
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with users', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminUsers
          load={spy}
          remove={spy}
          requestMagicLink={spy}
          create={spy}
          update={spy}
          adminUsersState={{
            ...initialState,
            users: [
              { id: '1', name: 'name', email: 'test@example.com' },
              { id: '2', name: 'another', email: 'another@example.com' },
              { id: '3', name: 'a third', email: 'three@example.com' },
            ],
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'testUser', admin: true },
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with requesting=true', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminUsers
          load={spy}
          remove={spy}
          requestMagicLink={spy}
          create={spy}
          update={spy}
          adminUsersState={{
            ...initialState,
            requesting: true,
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'testUser', admin: true },
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with requestError', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminUsers
          load={spy}
          remove={spy}
          requestMagicLink={spy}
          create={spy}
          update={spy}
          adminUsersState={{
            ...initialState,
            requestError: { error: 'some_error', errorMessage: 'Something went wrong' },
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'testUser', admin: true },
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with removing=true', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminUsers
          load={spy}
          remove={spy}
          requestMagicLink={spy}
          create={spy}
          update={spy}
          adminUsersState={{
            ...initialState,
            removing: true,
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'testUser', admin: true },
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with removeError', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminUsers
          load={spy}
          remove={spy}
          requestMagicLink={spy}
          create={spy}
          update={spy}
          adminUsersState={{
            ...initialState,
            removeError: { error: 'some_error', errorMessage: 'Something went wrong' },
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'testUser', admin: true },
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with creating=true', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminUsers
          load={spy}
          remove={spy}
          requestMagicLink={spy}
          create={spy}
          update={spy}
          adminUsersState={{
            ...initialState,
            creating: true,
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'testUser', admin: true },
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with createError', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminUsers
          load={spy}
          remove={spy}
          requestMagicLink={spy}
          create={spy}
          update={spy}
          adminUsersState={{
            ...initialState,
            createError: { error: 'some_error', errorMessage: 'Something went wrong' },
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'testUser', admin: true },
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with updating=true', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminUsers
          load={spy}
          remove={spy}
          requestMagicLink={spy}
          create={spy}
          update={spy}
          adminUsersState={{
            ...initialState,
            updating: true,
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'testUser', admin: true },
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with updateError', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminUsers
          load={spy}
          remove={spy}
          requestMagicLink={spy}
          create={spy}
          update={spy}
          adminUsersState={{
            ...initialState,
            updateError: { error: 'some_error', errorMessage: 'Something went wrong' },
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'testUser', admin: true },
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
