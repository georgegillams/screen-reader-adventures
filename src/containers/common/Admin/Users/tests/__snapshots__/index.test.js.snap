// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminUsers /> should render correctly - index 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="skeleton__outer skeleton__section"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="skeleton__outer skeleton__card--compact"
      style="margin-top: 1.2rem;"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="skeleton__outer skeleton__card--compact"
      style="margin-top: 1.2rem;"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="skeleton__outer skeleton__card--compact"
      style="margin-top: 1.2rem;"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="skeleton__outer skeleton__card--compact"
      style="margin-top: 1.2rem;"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
  </div>
</div>
`;

exports[`<AdminUsers /> should render correctly with admin user 1`] = `
<div>
  <div>
    <a
      class="text-link text-link--text-link text-link--no-padding text-link--inline page-title__link"
      href="/admin"
    >
      ⇠ Admin
    </a>
    <h1
      class="page-title__heading page-title__heading--with-link"
    >
      Admin users
    </h1>
  </div>
  <div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Reload users
      </span>
    </button>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Show filters
      </span>
    </button>
  </div>
  <div
    class="split-detail-view__outer admin-users__split-view"
  >
    <div
      class="split-detail-view__list-view"
    >
      <div>
        <a
          class="card admin-users__card card--atomic card--not-disabled"
          href="/admin/users?highlight=new"
        >
          <div
            class="card__outer-container"
          >
            <div
              class="card__background"
            />
            <div
              class="card__content-container"
            >
              <div
                class="split-detail-item__content"
              >
                <div
                  class="subsection__outer"
                >
                  <h3
                    class="subsection__heading subsection__heading--no-padding"
                    id="new-+"
                  >
                    New +
                  </h3>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AdminUsers /> should render correctly with createError 1`] = `
<div>
  <div>
    <a
      class="text-link text-link--text-link text-link--no-padding text-link--inline page-title__link"
      href="/admin"
    >
      ⇠ Admin
    </a>
    <h1
      class="page-title__heading page-title__heading--with-link"
    >
      Admin users
    </h1>
  </div>
  <div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Reload users
      </span>
    </button>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Show filters
      </span>
    </button>
  </div>
  <div
    class="split-detail-view__outer admin-users__split-view"
  >
    <div
      class="split-detail-view__list-view"
    >
      <div>
        <a
          class="card admin-users__card card--atomic card--not-disabled"
          href="/admin/users?highlight=new"
        >
          <div
            class="card__outer-container"
          >
            <div
              class="card__background"
            />
            <div
              class="card__content-container"
            >
              <div
                class="split-detail-item__content"
              >
                <div
                  class="subsection__outer"
                >
                  <h3
                    class="subsection__heading subsection__heading--no-padding"
                    id="new-+"
                  >
                    New +
                  </h3>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AdminUsers /> should render correctly with creating=true 1`] = `
<div>
  <div>
    <a
      class="text-link text-link--text-link text-link--no-padding text-link--inline page-title__link"
      href="/admin"
    >
      ⇠ Admin
    </a>
    <h1
      class="page-title__heading page-title__heading--with-link"
    >
      Admin users
    </h1>
  </div>
  <div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Reload users
      </span>
    </button>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Show filters
      </span>
    </button>
  </div>
  <div
    class="split-detail-view__outer admin-users__split-view"
  >
    <div
      class="split-detail-view__list-view"
    >
      <div>
        <a
          class="card admin-users__card card--atomic card--not-disabled"
          href="/admin/users?highlight=new"
        >
          <div
            class="card__outer-container"
          >
            <div
              class="card__background"
            />
            <div
              class="card__content-container"
            >
              <div
                class="split-detail-item__content"
              >
                <div
                  class="subsection__outer"
                >
                  <h3
                    class="subsection__heading subsection__heading--no-padding"
                    id="new-+"
                  >
                    New +
                  </h3>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AdminUsers /> should render correctly with initial state 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="skeleton__outer skeleton__section"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="skeleton__outer skeleton__card--compact"
      style="margin-top: 1.2rem;"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="skeleton__outer skeleton__card--compact"
      style="margin-top: 1.2rem;"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="skeleton__outer skeleton__card--compact"
      style="margin-top: 1.2rem;"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="skeleton__outer skeleton__card--compact"
      style="margin-top: 1.2rem;"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
  </div>
</div>
`;

exports[`<AdminUsers /> should render correctly with loadError 1`] = `
<div>
  <div>
    <a
      class="text-link text-link--text-link text-link--no-padding text-link--inline page-title__link"
      href="/admin"
    >
      ⇠ Admin
    </a>
    <h1
      class="page-title__heading page-title__heading--with-link"
    >
      Admin users
    </h1>
  </div>
  <div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Reload users
      </span>
    </button>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Show filters
      </span>
    </button>
  </div>
  <span
    class="paragraph"
  >
    <span
      class="error"
    >
      Could not load users
      <br />
      Something went wrong
    </span>
  </span>
  <div
    class="split-detail-view__outer admin-users__split-view"
  >
    <div
      class="split-detail-view__list-view"
    >
      <div>
        <a
          class="card admin-users__card card--atomic card--not-disabled"
          href="/admin/users?highlight=new"
        >
          <div
            class="card__outer-container"
          >
            <div
              class="card__background"
            />
            <div
              class="card__content-container"
            >
              <div
                class="split-detail-item__content"
              >
                <div
                  class="subsection__outer"
                >
                  <h3
                    class="subsection__heading subsection__heading--no-padding"
                    id="new-+"
                  >
                    New +
                  </h3>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AdminUsers /> should render correctly with loading=true 1`] = `
<div>
  <div>
    <a
      class="text-link text-link--text-link text-link--no-padding text-link--inline page-title__link"
      href="/admin"
    >
      ⇠ Admin
    </a>
    <h1
      class="page-title__heading page-title__heading--with-link"
    >
      Admin users
    </h1>
  </div>
  <div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy button__outer--disabled admin-users__button"
      disabled=""
      type="button"
    >
      <span
        class="button__child button__child--loading"
      >
        Reload users
      </span>
      <div
        aria-hidden="true"
        class="button__loading-spinner"
      >
        <div
          class="spinner__outer"
        >
          <div
            class="spinner__inner"
          />
          <div
            class="spinner__inner"
          />
          <div
            class="spinner__inner"
          />
          <div
            class="spinner__inner"
          />
          <div
            class="spinner__inner"
          />
          <div
            class="spinner__inner"
          />
          <div
            class="spinner__inner"
          />
          <div
            class="spinner__inner"
          />
        </div>
      </div>
    </button>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Show filters
      </span>
    </button>
  </div>
  <div
    class="split-detail-view__outer admin-users__split-view"
  >
    <div
      class="split-detail-view__list-view"
    >
      <div>
        <a
          class="card admin-users__card card--atomic card--not-disabled"
          href="/admin/users?highlight=new"
        >
          <div
            class="card__outer-container"
          >
            <div
              class="card__background"
            />
            <div
              class="card__content-container"
            >
              <div
                class="split-detail-item__content"
              >
                <div
                  class="subsection__outer"
                >
                  <h3
                    class="subsection__heading subsection__heading--no-padding"
                    id="new-+"
                  >
                    New +
                  </h3>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AdminUsers /> should render correctly with removeError 1`] = `
<div>
  <div>
    <a
      class="text-link text-link--text-link text-link--no-padding text-link--inline page-title__link"
      href="/admin"
    >
      ⇠ Admin
    </a>
    <h1
      class="page-title__heading page-title__heading--with-link"
    >
      Admin users
    </h1>
  </div>
  <div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Reload users
      </span>
    </button>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Show filters
      </span>
    </button>
  </div>
  <div
    class="split-detail-view__outer admin-users__split-view"
  >
    <div
      class="split-detail-view__list-view"
    >
      <div>
        <a
          class="card admin-users__card card--atomic card--not-disabled"
          href="/admin/users?highlight=new"
        >
          <div
            class="card__outer-container"
          >
            <div
              class="card__background"
            />
            <div
              class="card__content-container"
            >
              <div
                class="split-detail-item__content"
              >
                <div
                  class="subsection__outer"
                >
                  <h3
                    class="subsection__heading subsection__heading--no-padding"
                    id="new-+"
                  >
                    New +
                  </h3>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AdminUsers /> should render correctly with removing=true 1`] = `
<div>
  <div>
    <a
      class="text-link text-link--text-link text-link--no-padding text-link--inline page-title__link"
      href="/admin"
    >
      ⇠ Admin
    </a>
    <h1
      class="page-title__heading page-title__heading--with-link"
    >
      Admin users
    </h1>
  </div>
  <div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Reload users
      </span>
    </button>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Show filters
      </span>
    </button>
  </div>
  <div
    class="split-detail-view__outer admin-users__split-view"
  >
    <div
      class="split-detail-view__list-view"
    >
      <div>
        <a
          class="card admin-users__card card--atomic card--not-disabled"
          href="/admin/users?highlight=new"
        >
          <div
            class="card__outer-container"
          >
            <div
              class="card__background"
            />
            <div
              class="card__content-container"
            >
              <div
                class="split-detail-item__content"
              >
                <div
                  class="subsection__outer"
                >
                  <h3
                    class="subsection__heading subsection__heading--no-padding"
                    id="new-+"
                  >
                    New +
                  </h3>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AdminUsers /> should render correctly with requestError 1`] = `
<div>
  <div>
    <a
      class="text-link text-link--text-link text-link--no-padding text-link--inline page-title__link"
      href="/admin"
    >
      ⇠ Admin
    </a>
    <h1
      class="page-title__heading page-title__heading--with-link"
    >
      Admin users
    </h1>
  </div>
  <div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Reload users
      </span>
    </button>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Show filters
      </span>
    </button>
  </div>
  <div
    class="split-detail-view__outer admin-users__split-view"
  >
    <div
      class="split-detail-view__list-view"
    >
      <div>
        <a
          class="card admin-users__card card--atomic card--not-disabled"
          href="/admin/users?highlight=new"
        >
          <div
            class="card__outer-container"
          >
            <div
              class="card__background"
            />
            <div
              class="card__content-container"
            >
              <div
                class="split-detail-item__content"
              >
                <div
                  class="subsection__outer"
                >
                  <h3
                    class="subsection__heading subsection__heading--no-padding"
                    id="new-+"
                  >
                    New +
                  </h3>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AdminUsers /> should render correctly with requesting=true 1`] = `
<div>
  <div>
    <a
      class="text-link text-link--text-link text-link--no-padding text-link--inline page-title__link"
      href="/admin"
    >
      ⇠ Admin
    </a>
    <h1
      class="page-title__heading page-title__heading--with-link"
    >
      Admin users
    </h1>
  </div>
  <div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Reload users
      </span>
    </button>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Show filters
      </span>
    </button>
  </div>
  <div
    class="split-detail-view__outer admin-users__split-view"
  >
    <div
      class="split-detail-view__list-view"
    >
      <div>
        <a
          class="card admin-users__card card--atomic card--not-disabled"
          href="/admin/users?highlight=new"
        >
          <div
            class="card__outer-container"
          >
            <div
              class="card__background"
            />
            <div
              class="card__content-container"
            >
              <div
                class="split-detail-item__content"
              >
                <div
                  class="subsection__outer"
                >
                  <h3
                    class="subsection__heading subsection__heading--no-padding"
                    id="new-+"
                  >
                    New +
                  </h3>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AdminUsers /> should render correctly with updateError 1`] = `
<div>
  <div>
    <a
      class="text-link text-link--text-link text-link--no-padding text-link--inline page-title__link"
      href="/admin"
    >
      ⇠ Admin
    </a>
    <h1
      class="page-title__heading page-title__heading--with-link"
    >
      Admin users
    </h1>
  </div>
  <div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Reload users
      </span>
    </button>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Show filters
      </span>
    </button>
  </div>
  <div
    class="split-detail-view__outer admin-users__split-view"
  >
    <div
      class="split-detail-view__list-view"
    >
      <div>
        <a
          class="card admin-users__card card--atomic card--not-disabled"
          href="/admin/users?highlight=new"
        >
          <div
            class="card__outer-container"
          >
            <div
              class="card__background"
            />
            <div
              class="card__content-container"
            >
              <div
                class="split-detail-item__content"
              >
                <div
                  class="subsection__outer"
                >
                  <h3
                    class="subsection__heading subsection__heading--no-padding"
                    id="new-+"
                  >
                    New +
                  </h3>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AdminUsers /> should render correctly with updating=true 1`] = `
<div>
  <div>
    <a
      class="text-link text-link--text-link text-link--no-padding text-link--inline page-title__link"
      href="/admin"
    >
      ⇠ Admin
    </a>
    <h1
      class="page-title__heading page-title__heading--with-link"
    >
      Admin users
    </h1>
  </div>
  <div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Reload users
      </span>
    </button>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Show filters
      </span>
    </button>
  </div>
  <div
    class="split-detail-view__outer admin-users__split-view"
  >
    <div
      class="split-detail-view__list-view"
    >
      <div>
        <a
          class="card admin-users__card card--atomic card--not-disabled"
          href="/admin/users?highlight=new"
        >
          <div
            class="card__outer-container"
          >
            <div
              class="card__background"
            />
            <div
              class="card__content-container"
            >
              <div
                class="split-detail-item__content"
              >
                <div
                  class="subsection__outer"
                >
                  <h3
                    class="subsection__heading subsection__heading--no-padding"
                    id="new-+"
                  >
                    New +
                  </h3>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AdminUsers /> should render correctly with user 1`] = `
<div>
  <div
    class="section__outer"
  >
    <h2
      class="section__heading"
      id="admin-only"
    >
      Admin only
    </h2>
    <span
      class="paragraph"
    >
      You need to be logged in with an admin account to view this content.
      <br />
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/login"
      >
        Got a different admin account? Log in here.
      </a>
    </span>
  </div>
</div>
`;

exports[`<AdminUsers /> should render correctly with users 1`] = `
<div>
  <div>
    <a
      class="text-link text-link--text-link text-link--no-padding text-link--inline page-title__link"
      href="/admin"
    >
      ⇠ Admin
    </a>
    <h1
      class="page-title__heading page-title__heading--with-link"
    >
      Admin users
    </h1>
  </div>
  <div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Reload users
      </span>
    </button>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-users__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Show filters
      </span>
    </button>
  </div>
  <span
    class="paragraph admin-analytics__count"
  >
    Showing 
    3
     of 
    3
     users
  </span>
  <div
    class="split-detail-view__outer admin-users__split-view"
  >
    <div
      class="split-detail-view__list-view"
    >
      <div>
        <a
          class="card admin-users__card card--atomic card--not-disabled"
          href="/admin/users?highlight=new"
        >
          <div
            class="card__outer-container"
          >
            <div
              class="card__background"
            />
            <div
              class="card__content-container"
            >
              <div
                class="split-detail-item__content"
              >
                <div
                  class="subsection__outer"
                >
                  <h3
                    class="subsection__heading subsection__heading--no-padding"
                    id="new-+"
                  >
                    New +
                  </h3>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          class="card admin-users__card card--atomic card--not-disabled"
          href="/admin/users?highlight=1"
          id="1"
        >
          <div
            class="card__outer-container"
          >
            <div
              class="card__background"
            />
            <div
              class="card__content-container"
            >
              <div
                class="split-detail-item__content"
              >
                <div
                  class="subsection__outer"
                >
                  <h3
                    class="subsection__heading subsection__heading--no-padding"
                    id="name"
                  >
                    name
                  </h3>
                  <span
                    class="paragraph"
                  >
                    <br />
                    id: 
                    1
                  </span>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          class="card admin-users__card card--atomic card--not-disabled"
          href="/admin/users?highlight=2"
          id="2"
        >
          <div
            class="card__outer-container"
          >
            <div
              class="card__background"
            />
            <div
              class="card__content-container"
            >
              <div
                class="split-detail-item__content"
              >
                <div
                  class="subsection__outer"
                >
                  <h3
                    class="subsection__heading subsection__heading--no-padding"
                    id="another"
                  >
                    another
                  </h3>
                  <span
                    class="paragraph"
                  >
                    <br />
                    id: 
                    2
                  </span>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          class="card admin-users__card card--atomic card--not-disabled"
          href="/admin/users?highlight=3"
          id="3"
        >
          <div
            class="card__outer-container"
          >
            <div
              class="card__background"
            />
            <div
              class="card__content-container"
            >
              <div
                class="split-detail-item__content"
              >
                <div
                  class="subsection__outer"
                >
                  <h3
                    class="subsection__heading subsection__heading--no-padding"
                    id="a-third"
                  >
                    a third
                  </h3>
                  <span
                    class="paragraph"
                  >
                    <br />
                    id: 
                    3
                  </span>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
